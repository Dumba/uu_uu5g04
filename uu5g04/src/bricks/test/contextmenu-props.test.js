/**
 * Copyright (C) 2019 Unicorn a.s.
 *
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 *
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

//@@viewOn:imports
import UU5 from "uu5g04";
import "uu5g04-bricks";
//@@viewOff:imports

const { mount, shallow, wait } = UU5.Test.Tools;

/**
 * This is a created component for the Allow Tags test.
 * It is tested that a self-created component can be inserted into the accordion under its own brand.
 */
const MyAllowTagsComponents = UU5.Common.VisualComponent.create({
  mixins: [UU5.Common.BaseMixin],
  statics: { tagName: "UU5.Example.MyCompButton", classNames: { main: "mytr" } },
  render() {
    return <UU5.Example.MyCompButton {...this.getMainPropsToPass()} />;
  }
});

const CONFIG = {
  mixins: [
    "UU5.Common.BaseMixin",
    "UU5.Common.ElementaryMixin",
    "UU5.Common.SectionMixin",
    "UU5.Common.LevelMixin",
    "UU5.Common.ContentMixin",
    "UU5.Common.CcrWriterMixin",
    "UU5.Common.PureRenderMixin",
    "UU5.Common.NestingLevelMixin"
  ],
  props: {
    shown: {
      values: [true, false]
    },
    // parentElement: {}, - Expression from DEV - Do not test this props.
    allowTags: {
      allowTagsArray: ["UU5.Example.MyCompButton"]
    },
    compactSubmenu: {
      values: [true, false]
    }
  },
  requiredProps: {
    children: [<UU5.Bricks.ContextMenu.Item id={"uuID2"} label="Ráno" icon="mdi-weather-sunset-up" />]
  },
  opt: {
    shallowOpt: {
      disableLifecycleMethods: false
    }
  }
};

describe(`UU5.Bricks.ContextMenu`, () => {
  UU5.Test.Tools.testProperties(UU5.Bricks.ContextMenu, CONFIG);
});

describe(`UU5.Bricks.ContextMenu docKit examples`, () => {
  it(`UU5.Bricks.ContextMenu myComponent from allowTags is used`, () => {
    const wrapper = shallow(
      <UU5.Bricks.ContextMenu
        id={"uuID"}
        shown={true}
        header="Header"
        footer="Footer"
        allowTags={["UU5.Example.MyCompButton"]}
      >
        <UU5.Bricks.ContextMenu.Item id={"uuID2"} label="Ráno" icon="mdi-weather-sunset-up" space />
        <MyAllowTagsComponents id={"allowId"} content={"allowTagsContent"} />
      </UU5.Bricks.ContextMenu>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it(`UU5.Bricks.ContextMenu example make snapshot`, () => {
    const wrapper = shallow(
      <UU5.Bricks.ContextMenu id={"uuID"} shown={true} header="Header" footer="Footer">
        <UU5.Bricks.ContextMenu.Item id={"uuID2"} label="Ráno" icon="mdi-weather-sunset-up" space />
        <UU5.Bricks.ContextMenu.Item id={"uuID3"} label="Dopoledne" space />
        <UU5.Bricks.ContextMenu.Item id={"uuID4"} label="V poledne" icon="mdi-weather-sunny" space disabled />
        <UU5.Bricks.ContextMenu.Item id={"uuID5"} label="Odpoledne" space />
        <UU5.Bricks.ContextMenu.Item id={"uuID6"} label="Večer" icon="mdi-weather-sunset" />
        <UU5.Bricks.ContextMenu.Item id={"uuID7"} divider />
        <UU5.Bricks.ContextMenu.Item id={"uuID10"} label="Úplně jiné" disabled />
      </UU5.Bricks.ContextMenu>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

const getContextMenu = props => (
  <UU5.Bricks.ContextMenu {...props}>
    <UU5.Bricks.ContextMenu.Item label="Level 0" space>
      <UU5.Bricks.ContextMenu.Item label="Level 1" space>
        <UU5.Bricks.ContextMenu.Item label="Level 2" space />
      </UU5.Bricks.ContextMenu.Item>
      <UU5.Bricks.ContextMenu.Item label="Level 1 Standard Item" space />
    </UU5.Bricks.ContextMenu.Item>
  </UU5.Bricks.ContextMenu>
);

describe(`UU5.Bricks.ContextMenu functionality`, () => {
  it(`UU5.Bricks.ContextMenu compactSubmenu`, () => {
    const origGetScreenSize = UU5.Common.Tools.getScreenSize;

    UU5.Common.Tools.getScreenSize = () => "l";
    let wrapper = mount(getContextMenu({ compactSubmenu: "xs s" }));
    expect(
      wrapper.findWhere(
        node =>
          node.instance() &&
          node.instance().getTagName &&
          node.instance().getTagName() === "UU5.Bricks.CompactContextMenu"
      ).length
    ).toBe(0);

    UU5.Common.Tools.getScreenSize = () => "s";
    wrapper = mount(getContextMenu({ compactSubmenu: "xs s" }));
    expect(
      wrapper.findWhere(
        node =>
          node.instance() &&
          node.instance().getTagName &&
          node.instance().getTagName() === "UU5.Bricks.CompactContextMenu"
      ).length
    ).toBe(1);

    wrapper.instance().open();
    wrapper.update();
    let item = wrapper.findWhere(node => node.hasClass("uu5-bricks-link") && node.text() === "Level 0");
    item.simulate("click");
    wrapper.update();

    expect(wrapper.findWhere(node => node.hasClass("uu5-bricks-link") && node.text() === "Back").length).toBe(1);
    item = wrapper.findWhere(node => node.hasClass("uu5-bricks-link") && node.text() === "Level 1");
    item.simulate("click");
    wrapper.update();

    let backButton = wrapper.findWhere(node => node.hasClass("uu5-bricks-link") && node.text() === "Back");
    expect(wrapper.findWhere(node => node.hasClass("uu5-bricks-link") && node.text() === "Level 2").length).toBe(1);
    backButton.simulate("click");
    wrapper.update();

    item = wrapper.findWhere(node => node.hasClass("uu5-bricks-link") && node.text() === "Level 1 Standard Item");
    item.simulate("click");
    wrapper.update();
    expect(wrapper.instance().isOpen()).toBeFalsy();

    UU5.Common.Tools.getScreenSize = origGetScreenSize;
  });
});
