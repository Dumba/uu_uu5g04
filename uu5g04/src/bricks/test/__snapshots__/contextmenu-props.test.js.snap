// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.ContextMenu allowTags=undefined 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu ccrKey="<UU5.Parrot.Page ccrKey='UU5.Parrot.Page' />" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu checkGrammar=false 1`] = `
<Popover
  checkGrammar={false}
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu checkGrammar=true 1`] = `
<Popover
  checkGrammar={true}
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu checkHighlight=false 1`] = `
<Popover
  checkHighlight={false}
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu checkHighlight=true 1`] = `
<Popover
  checkHighlight={true}
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu checkSpaces=false 1`] = `
<Popover
  checkSpaces={false}
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu checkSpaces=true 1`] = `
<Popover
  checkSpaces={true}
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu className="my-classname" 1`] = `
<Popover
  className="uu5-bricks-context-menu my-classname"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu compactSubmenu=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu compactSubmenu=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu content="This is Content from ContentMixin" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  content="This is Content from ContentMixin"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu content=12345 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  content={12345}
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu controlled=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu controlled=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu default props 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu disabled=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu disabled=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  disabled={true}
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu docKit examples UU5.Bricks.ContextMenu example make snapshot 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  footer="Footer"
  header="Header"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
  shown={true}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu docKit examples UU5.Bricks.ContextMenu myComponent from allowTags is used 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  footer="Footer"
  header="Header"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
  shown={true}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu dynamic=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu dynamic=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  dynamic={true}
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu footer={"type":"div","key":null,"ref":null,"props":{"children":"Footer of section"},"_owner":null,"_store":{}} 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  footer={
    <div>
      Footer of section
    </div>
  }
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu header={"type":"div","key":null,"ref":null,"props":{"children":"Header of section"},"_owner":null,"_store":{}} 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  header={
    <div>
      Header of section
    </div>
  }
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu hidden=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu hidden=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  hidden={true}
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu id="myId" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="myId-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu ignoreHTML=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu ignoreHTML=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level="0" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level="0"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level="1" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level="1"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level="2" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level="2"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level="3" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level="3"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level="4" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level="4"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level="5" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level="5"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level="6" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level="6"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level=0 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level={0}
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level=1 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level={1}
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level=2 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level={2}
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level=3 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level={3}
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level=4 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level={4}
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level=5 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level={5}
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu level=6 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  level={6}
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu mainAttrs=undefined 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu name="myName" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  name="myName-inner"
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="bigBox" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBox"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="bigBoxCollection" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="box" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="box"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="boxCollection" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="boxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="container" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="inline" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="page" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="smallBox" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="smallBoxCollection" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu nestingLevel="spa" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu noIndex=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu noIndex=true 1`] = `
<Popover
  className="uu5-bricks-context-menu uu5-noindex"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu pureRender=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu pureRender=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  pureRender={true}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu selected=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu selected=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
  selected={true}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu shown=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu shown=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
  shown={true}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu style="padding:8px; borderRadius: 2px" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
  style={
    Object {
      "borderRadius": "2px",
      "padding": "8px",
    }
  }
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu style={"backgroundColor":"red","color":"blue"} 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "color": "blue",
    }
  }
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu textCorrector=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
  textCorrector={false}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu textCorrector=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
  textCorrector={true}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu tooltip="My Tooltip" 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={
    Object {
      "title": "My Tooltip",
    }
  }
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
  tooltip="My Tooltip"
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu underline=false 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;

exports[`UU5.Bricks.ContextMenu underline=true 1`] = `
<Popover
  className="uu5-bricks-context-menu"
  id="uuID-inner"
  mainAttrs={Object {}}
  nestingLevel="bigBoxCollection"
  parentElement={null}
  ref_={[Function]}
>
  <ScreenSize>
    <Component />
  </ScreenSize>
</Popover>
`;
